project_generator/tools/coide.py

```diff
@@ -72,59 +72,8 @@ def get_toolnames():
     def get_toolchain():
         return 'gcc_arm'
 
-    def _expand_data(self, old_data, new_data, attribute, group):
-        """ data expansion - coide needs filename and path separately """
-        # TODO: fix this for include/sources. Do we need it?
-        if group == 'Sources':
-            old_group = None
-        else:
-            old_group = group
-
-        for file in old_data[old_group]:
-            if file:
-                extension = file.split(".")[-1]
-                if not extension in self.file_types.keys():
-                    logger.debug("Filetype for file %s not recognized" % file)
-                    continue
-                new_file = {
-                    '@path': file, '@name': basename(file), '@type': str(self.file_types[extension.lower()])
-                }
-                new_data['groups'][group].append(new_file)
-
-    def _get_groups(self):
-        """ Get all groups defined """
-        groups = []
-        for attribute in SOURCE_KEYS:
-            for k, v in self.workspace[attribute].items():
-                if k == None:
-                    k = 'Sources'
-                if k not in groups:
-                    groups.append(k)
-            for k, v in self.workspace['include_files'].items():
-                if k == None:
-                    k = 'Includes'
-                if k not in groups:
-                    groups.append(k)
-        return groups
-
-    def _iterate(self, data, expanded_data):
-        """ _Iterate through all data, store the result expansion in extended dictionary """
-        for attribute in SOURCE_KEYS:
-            for k, v in data[attribute].items():
-                if k == None:
-                    group = 'Sources'
-                else:
-                    group = k
-                self._expand_data(data[attribute], expanded_data, attribute, group)
-        for k, v in data['include_files'].items():
-            if k == None:
-                group = 'Includes'
-            else:
-                group = k
-            self._expand_data(data['include_files'], expanded_data, attribute, group)
-
-        # sort groups
-        expanded_data['groups'] = OrderedDict(sorted(expanded_data['groups'].items(), key=lambda t: t[0]))
+    def _expand_one_file(self, source, new_data, extension):
+        return {'@path': source, '@name': basename(source), '@type': str(self.file_types[extension.lower()])}
```

Comment:
What do you think about the debug message when file type is not recognized that was there previously? I found it useful to have as file types are limited in IDE (they have own values per a file type).  otherwise we end up in the error with the traceback, not useful for a user?